src/internalEnforcer.ts

```diff
@@ -44,6 +45,34 @@ export class InternalEnforcer extends CoreEnforcer {
     return this.model.addPolicy(sec, ptype, rule);
   }
 
+  // addPolicies adds rules to the current policy.
+  // removePolicies removes rules from the current policy.
+  public async addPoliciesInternal(sec: string, ptype: string, rules: string[][]): Promise<boolean> {
+    for (const rule of rules) {
+      if (this.model.hasPolicy(sec, ptype, rule)) {
+        return false;
+      }
+    }
+
+    const batchAdapter = this.adapter as BatchAdapter;
+    if (batchAdapter && this.autoSave) {
+      try {
+        await batchAdapter.addPolicies(sec, ptype, rules);
```

Comment:
I believe adapter should also give the information if the policies have been inserted (`true` OR `false`), no? If yes then we still need to take into account of the return value.

Reply:
@GopherJ, I have followed the ```addPolicyInternal()``` and ```removePolicyInternal()``` implementation, they also does not take care of return values either.

Reply:
@nodece  How do you think? does `no error throwed` means policy has been inserted/removed? I have doubt actually because sometimes users may just give some bad input and the adapter can return `false` in this case but often bad input doesn't mean there is an error, we just ignored the input.

If we don't take into account of this, it's also ok but we will rely on the implementation of adapters, adapter can only return true because false wont make sense anymore.

Reply:
https://github.com/casbin-rs/diesel-adapter/blob/ac1d2fdc1c0e37c60dcb86297ef65006ace96114/src/adapter.rs#L297

Like here, if the `field_index` is greater than 5, I would like the adapter return `fase` and then I return back this value to user.

But if we don't take into account of `false` value, then user will finally get `true` which is confusing.

Reply:
The ```casbin-core``` adapter interface also has to be changed, if this changes, hence, it solely depends upon which implementation is clean and necessary enough to make changes which is matter of discussion.

Reply:
> @nodece How do you think? does `no error throwed` means policy has been inserted/removed? I have doubt because sometimes users may just give some bad input and the adapter can return `false` in this case but often bad input doesn't mean there is an error, we just ignored the input.
> 
> If we don't take into account of this, it's also ok but we will rely on the implementation of adapters, adapter can only return true because false wont make sense anymore.

In node-casbin, `adapter` returns `Promise<void>`. 

A `Promise` with either a resolved or rejected value: 
- resolve(value) — if the job finished successfully, with result value.
- reject(error) — if an error occurred, error is the error object.

When policy has been inserted/removed, `adapter` call `Promise.resolve`, otherwise, throw error to `Promise.reject`.

Reply:
So, is this issue resolved?

Reply:
I think current implement is correct.

Reply:
ok I didn't notice that the interface is `Promise<void>`, in this case I think it's fine